### ISSUE TITLE
Line mark with point overlay should use use strokeWidth encoding instead of size

### ISSUE THREAD

**Issue by marcprux** [vega/vega-lite#6841]
An undocumented feature of area marks with point & line overlays is that the encoding channels will apply to the secondary point/line channel. So the following spec will apply the "size" to the points:

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.14.0.json",
  "data": {"url": "data/stocks.csv"},
  "transform": [{"filter": "datum.symbol==='AAPL'"}],
  "mark": {"type": "area", "point": true},
  "encoding": {
    "x": {"field": "date", "type": "temporal"},
    "y": {"field": "price", "type": "quantitative"},
    "size": { "value": 44 }
  }
}
```

<img width="278" alt="Screen Shot 2020-09-01 at 16 51 21" src="https://user-images.githubusercontent.com/659086/91904709-5fbf7b00-ec73-11ea-97de-36b540c759b3.png">

However, if you add in a line overlay as well, the "size" encoding is applied *both* to the point size and the line width:

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.14.0.json",
  "data": {"url": "data/stocks.csv"},
  "transform": [{"filter": "datum.symbol==='AAPL'"}],
  "mark": {"type": "area", "point": true, "line": true},
  "encoding": {
    "x": {"field": "date", "type": "temporal"},
    "y": {"field": "price", "type": "quantitative"},
    "size": { "value": 44 }
  }
}
```

<img width="320" alt="Screen Shot 2020-09-01 at 16 52 51" src="https://user-images.githubusercontent.com/659086/91904825-95646400-ec73-11ea-82a7-f788b8ae9275.png">

This is especially pernicious, since the line mark interprets the `size` constant linearly, but the point mark interprets the `size` constant as the square root of the total size, so there is no value that would ever be useful for both line and point.

One solution might be to continue to apply the "size" to the `point` overlay mark, but apply the "strokeWidth" encoding to the `line` overlay mark:

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.14.0.json",
  "data": {"url": "data/stocks.csv"},
  "transform": [{"filter": "datum.symbol==='AAPL'"}],
  "mark": {"type": "area", "point": true, "line": true},
  "encoding": {
    "x": {"field": "date", "type": "temporal"},
    "y": {"field": "price", "type": "quantitative"},
    "size": { "value": 44 },
    "strokeWidth": { "value": 10 }
  }
}
```

This would allow the `area` mark to enable a lot more interesting encoding combinations.

**Comment by marcprux** [comment]
I notice that both `strokeWidth` and `size` encodings affect the thickness of a `line` mark, but `size` takes precedence over `strokeWidth`. A very simple solution to this issue might be to just make `strokeWidth` have higher precedence than `size`, which also makes intuitive sense. The `size` encoding would then be free to be passed along to the `point` overlay.

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.14.0.json",
  "data": {"url": "data/stocks.csv"},
  "transform": [{"filter": "datum.symbol==='AAPL'"}],
  "mark": {"type": "line"},
  "encoding": {
    "x": {"field": "date", "type": "temporal"},
    "y": {"field": "price", "type": "quantitative"},
    "size": { "value": 10 },
    "strokeWidth": { "value": 3 }
  }
}
```

<img width="280" alt="Screen Shot 2020-09-01 at 20 33 51" src="https://user-images.githubusercontent.com/659086/91918953-745f3b80-ec92-11ea-8f6d-ee981edc1085.png">

**Comment by domoritz** [comment]
Yes, I agree it makes sense to have strokeWidth take precedence over size (similar to strokeColor and color).
